include/aws/crt/io/Uri.h

```diff
@@ -0,0 +1,102 @@
+#pragma once
+/*
+ * Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+#include <aws/crt/Types.h>
+
+#include <aws/io/uri.h>
+
+namespace Aws
+{
+    namespace Crt
+    {
+        namespace Io
+        {
+            /**
+             * Contains a URI used for networking application protocols. This type is move-only.
+             */
+            class AWS_CRT_CPP_API Uri final
+            {
+              public:
+                Uri() noexcept;
+                ~Uri();
+                /**
+                 * Parses `cursor` as a URI. Upon failure the bool() operator will return false and LastError()
+                 * will contain the errorCode.
+                 */
+                Uri(const ByteCursor &cursor, Allocator *allocator = DefaultAllocator()) noexcept;
+                /**
+                 * builds a URI from `builderOptions`. Upon failure the bool() operator will return false and
+                 * LastError() will contain the errorCode.
+                 */
+                Uri(aws_uri_builder_options &builderOptions, Allocator *allocator = DefaultAllocator()) noexcept;
+                Uri(const Uri &) = delete;
+                Uri &operator=(const Uri &) = delete;
+                Uri(Uri &&uri) noexcept;
+                Uri &operator=(Uri &&) noexcept;
```

Comment:
If you deleted the `operator=(&&)`, this would be an immutable class

not sure which is better, but whatever you choose, do the same with vanilla `operator=(&)`